lib/ansible/modules/cloud/vmware/vmware_guest_file_operation.py

```diff
@@ -0,0 +1,404 @@
+    vm_password = module.params['vm_password']
+
+    recurse = bool(module.params['directory']['recurse'])
+    operation = module.params["directory"]['operation']
+    path = module.params["directory"]['path']
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_manager = content.guestOperationsManager.fileManager
+    if operation == "create":
+        try:
+            file_manager.MakeDirectoryInGuest(vm=vm, auth=creds, directoryPath=path,
+                                              createParentDirectories=recurse)
+        except vim.fault.FileAlreadyExists:
+            result['changed'] = False
+            result['msg'] = "Guest directory %s already exist" % path
+        except vim.fault.GuestPermissionDenied:
+            module.fail_json(msg="Permission denied for path %s" % path, uuid=vm.summary.config.uuid)
+        except vim.fault.InvalidGuestLogin:
+            module.fail_json(msg="Invalid guest login for user %s" % vm_username, uuid=vm.summary.config.uuid)
+        # other exceptions
+        except Exception as e:
+            module.fail_json(msg="Failed to Create directory into Vm VMware exception:%s" % e,
+                             uuid=vm.summary.config.uuid)
+
+    if operation == "delete":
+        try:
+            file_manager.DeleteDirectoryInGuest(vm=vm, auth=creds, directoryPath=path,
+                                                recursive=recurse)
+        except vim.fault.FileNotFound:
+            result['changed'] = False
+            result['msg'] = "Guest directory %s not exists" % path
+        except vim.fault.FileFault as e:
+            module.fail_json(msg="FileFault:%s" % e.msg, uuid=vm.summary.config.uuid)
+        except vim.fault.GuestPermissionDenied:
+            module.fail_json(msg="Permission denied for path %s" % path, uuid=vm.summary.config.uuid)
+        except vim.fault.InvalidGuestLogin:
+            module.fail_json(msg="Invalid guest login for user %s" % vm_username, uuid=vm.summary.config.uuid)
+        # other exceptions
+        except Exception as e:
+            module.fail_json(msg="Failed to Delete directory into Vm VMware exception:%s" % e,
+                             uuid=vm.summary.config.uuid)
+
+    return result
+
+
+def fetch(module, content, vm):
+    result = {}
+    result['changed'] = True
+    result['uuid'] = vm.summary.config.uuid
+    vm_username = module.params['vm_username']
+    vm_password = module.params['vm_password']
+    dest = module.params["fetch"]['dest']
+    src = module.params['fetch']['src']
+    creds = vim.vm.guest.NamePasswordAuthentication(username=vm_username, password=vm_password)
+    file_manager = content.guestOperationsManager.fileManager
+
+    try:
+        fileTransferInfo = file_manager.InitiateFileTransferFromGuest(vm=vm, auth=creds,
+                                                                      guestFilePath=src)
+    except vim.fault.FileNotFound:
```

Comment:
It people are trying to download a non-exiting file, the execution should fail, because this is unexpected behaviour and should clearly be mentioned.

Reply:
@pdellaert could you have a look